gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/ConnectionPool.java

```diff
@@ -342,7 +320,7 @@ private boolean addConnectionIfUnderMaximum() {
             return false;
         }
 
-        announceAvailableConnection();
```

Comment:
When we return a connection to the pool in `ConnectionPool.returnConnection`, we use `announceAvailableConnection` and do not announce to all waiting threads whereas here on creation of a new connection we notify all waiting threads. Semantically whether a connection is being returned to the pool or a new connection is being created are same from the perspective of a waiting thread, hence, we should not have difference between notification behaviour.

Reply:
When a new Connection is created, `inFlight` count will be at 0, thus all waiting threads to see if they can pick up this new Connection.

When a Connection is returned to the pool, `inFlight` count is reduced by 1 from its current value, which if there are any threads waiting the Connection is currently at `maxSimulatenousUsage` because otherwise there wouldn't be threads waiting for it. That is why there is only need to notify a single thread here.